rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -780,6 +785,13 @@ module RustDataFlow implements InputSig<Location> {
         node1.asExpr() = tuple.getField(c.(TuplePositionContent).getPosition()) and
         node2.asExpr() = tuple
       )
+      or
+      exists(AssignmentExprCfgNode assignment, FieldExprCfgNode access |
+        assignment.getLhs() = access and
+        fieldTuplePositionContent(access, c) and
+        node1.asExpr() = assignment.getRhs() and
+        node2.asExpr() = access.getExpr()
+      )
```

Comment:
Should be `node2.(PostUpdateNode).getPreUpdateNode().asExpr()`. But I think we should make a helper predicate
```
private predicate tupleAssignment(Node node1, Node node2, TuplePositionContent c) {
  exists(AssignmentExprCfgNode assignment, FieldExprCfgNode access |
    assignment.getLhs() = access and
    fieldTuplePositionContent(access, c) and
    node1.asExpr() = assignment.getRhs() and
    node2.asExpr() = access.getExpr()
  )
}
```
Then the above becomes `tupleAssignment(node1, node2.(PostUpdateNode).getPreUpdateNode(), c)` and `clearsContent` becomes `tupleAssignment(_, n, cs.(SingletonContentSet).getContent())`